### PR TITLE
Add ARTIC ONT workflow

### PR DESCRIPTION
**Pull request by drpatelh**
This was inspired by the implementation in [ncov2019-artic-nf](https://github.com/connor-lab/ncov2019-artic-nf). When this is a little more finessed I will get in touch with Matt and ask how we can acknowledge all of the appropriate contributors to the credits. On the whole, I have re-written things with nf-core/modules in mind and tried to re-use as much of the code as possible that we are using for the Illumina implementation in this pipeline. DSL2!!!

I have also added a bunch of stuff on top like `QUAST`, `snpEff`, `Pangolin` and `MultiQC` (including more custom content for missing/filtered barcodes). I may also add in `mosdepth` for coverage QC but need to look into the MultiQC support for ARTIC tools which I believe may have been added recently.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -163,11 +186,36 @@ The Nextflow `-bg` flag launches Nextflow in the background, detached from your
...
+
+```bash
+export NXF_VER="20.11.0-edge"
+nextflow self-update
+```
+
+* If you would like to download and install a Nextflow `edge` release from scratch with minimal fuss:
+
+```bash
+export NXF_VER="20.12.0-edge"
```
Comment (heuermh) [inline docs/usage.md:214]: Should this version (`20.11.0-edge`) match that above (`20.12.0-edge`)?  Do we need to use a specific version here?  They will fall out of date over time

**Review by heuermh — CHANGES_REQUESTED** [review]
Minor doc-related changes requested, will attempt to run this on my own and may have additional suggestions

File: README.md [hunk 2]
```diff
@@ -77,27 +90,45 @@ Numerous QC and reporting steps are included in the pipeline in order to collate
...
 
+    * Typical command for Nanopore amplicon analysis:
+
+        ```bash
+        nextflow run nf-core/viralrecon \
+            --input samplesheet.csv \
+            --platform nanopore \
+            --genome 'MN908947.3' \
+            --primer_set_version 3 \
+            --fastq_dir ./fastq_pass/ \
```
[the remark this answers is not in the document]
Comment (drpatelh) [reply README.md:126]: It would work equally well without it. Just wanted to highlight that it's a directory really.

Comment (drpatelh) [inline README.md:126]:
```suggestion
            --fastq_dir fastq_pass/ \
```

File: README.md [hunk 3]
```diff
@@ -77,27 +90,45 @@ Numerous QC and reporting steps are included in the pipeline in order to collate
...
+            --fast5_dir ./fast5_pass/ \
```
Comment (drpatelh) [inline README.md:127]:
```suggestion
            --fast5_dir fast5_pass/ \
```

File: docs/usage.md (hunk 1 above)
Comment (drpatelh) [inline docs/usage.md:214]:
```suggestion
export NXF_VER="20.11.0-edge"
```

Comment (drpatelh) [reply docs/usage.md:214]: Eagle eyes!

File: docs/output.md [hunk 4]
```diff
@@ -1,58 +1,303 @@
...
+
+### Nanopore: Variant calling
+
+#### Nanopore: artic minion
+
+<details markdown="1">
+<summary>Output files</summary>
+
+* `nanopore/<CALLER>/`
+  * `*.consensus.fasta`: Consensus fasta file generated by arcic minion.
```
Comment (drpatelh) [inline docs/output.md:84]:
```suggestion
  * `*.consensus.fasta`: Consensus fasta file generated by artic minion.

```

File: docs/output.md [hunk 5]
```diff
@@ -1,58 +1,303 @@
...
+<summary>Output files</summary>
+
+* `nanopore/<CALLER>/pangolin/`
+  * `*.pangolin.csv`: Lineage analysis results from Pangolin.
+
+> **NB:** The value of `<CALLER>` in the output directory name above is determined by the `artic_minion_caller` parameter (Default: 'nanopolish').
+
+</details>
+
+Lineage analysis with [`Pangolin`](https://github.com/cov-lineages/pangolin)
```
[the remark this answers is not in the document]
Comment (drpatelh) [reply docs/output.md:219]: Yup will add it in next week 👍🏽

File: docs/output.md [hunk 6]
```diff
@@ -1,58 +1,303 @@
...
+  * [Illumina: SPAdes](#illumina-spades) *||* [Unicycler](#illumina-unicycler) *||* [minia](#illumina-minia) - Viral genome assembly
+    * [Illumina: BLAST](#illumina-blast) - Blast to reference assembly
+    * [Illumina: ABACAS](#illumina-abacas) - Order contigs according to reference genome
+    * [Illumina: PlasmidID](#illumina-plasmidid) - Assembly report and visualisation
+    * [Illumina: Assembly QUAST](#illumina-assembly-quast) - Assembly quality assessment
+    * [Illumina: Minimap2, seqwish, vg](#illumina-minimap2-seqwish-vg) - Call variants from induced genome variation graph
+    * [Illumina: Assembly SnpEff and SnpSift](#illumina-assembly-snpeff-and-snpsift) - Genetic variant annotation and functional effect prediction
+* [Illumina: Workflow reporting and genomes](#illumina-workflow-reporting-and-genomes)
+  * [Illumina: MultiQC](#illumina-multiqc) - Present QC for raw reads, alignment, assembly and variant calling
+  * [Illumina: Reference genome files](#illumina-reference-genome-files) - Saving reference genome indices/files
```
Comment (drpatelh) [inline docs/output.md:292]:
```suggestion
  * [Illumina: Reference genome files](#illumina-reference-genome-files) - Save reference genome indices/files

```

**Comment by drpatelh** [comment]
Ok done! I think 😁

**Review by heuermh — APPROVED** [review]

**Comment by drpatelh** [comment]
Thanks @heuermh !

**Pull request merged**